src/mlpack/methods/kde/kde_impl.hpp

```diff
@@ -569,6 +681,11 @@ serialize(Archive& ar, const unsigned int /* version */)
   ar & BOOST_SERIALIZATION_NVP(absError);
   ar & BOOST_SERIALIZATION_NVP(trained);
   ar & BOOST_SERIALIZATION_NVP(mode);
+  ar & BOOST_SERIALIZATION_NVP(monteCarlo);
+  ar & BOOST_SERIALIZATION_NVP(MCProb);
+  ar & BOOST_SERIALIZATION_NVP(initialSampleSize);
+  ar & BOOST_SERIALIZATION_NVP(MCEntryCoef);
+  ar & BOOST_SERIALIZATION_NVP(MCBreakCoef);
```

Comment:
Technically, this isn't reverse-compatible---KDE models made with earlier versions of mlpack won't have these members, so we should bump the Boost serialization version with a block like

```
//! Set the serialization version of the LSHSearch class.
BOOST_TEMPLATE_CLASS_VERSION(template<typename ...>,
    mlpack::kde::KDE<...>, 1);
```

Then, in `serialize()`, you can use the `version` parameter to detect if this is a version 0 archive (which will not have the Monte Carlo parameters) or a version 1 or newer archive (in which case it will).  You can take a look at `src/mlpack/methods/lsh/lsh_search_impl.hpp` for an example of how this might work. :+1:

Reply:
There's a small issue with using ```BOOST_TEMPLATE_CLASS_VERSION(...)``` [because of multi argument templates in macros](https://stackoverflow.com/questions/8942912/how-to-pass-multi-argument-templates-to-macros). Anyway I implemented using boost version struct and it works :)